airbyte-config/config-models/src/main/resources/types/StateType.yaml

```diff
@@ -0,0 +1,10 @@
+---
+"$schema": http://json-schema.org/draft-07/schema#
+"$id": https://github.com/airbytehq/airbyte/blob/master/airbyte-config/models/src/main/resources/types/StateType.yaml
+title: StateType
+description: State Types
+type: string
+enum:
+  - global
+  - stream
+  - legacy
```

Comment:
Can we add `notSet` here, I think it will be valuable.

Reply:
It is not clear the value it adds. I understand we currently consider `notSet` as `legacy` and we are should be defaulting new ones to `global` or `stream`.
With that in mind, `notSet` feels like a duplicate of `legacy` at the moment.

Reply:
I think this can be confusing for example:
- start: legacy state
- 1 sync is run with the new format: stream or global state
- 1 full reset is run, all the set will be deleted. The state will then be considered as a legacy state.

Reply:
As discussed, we can use `Optional<StateWrapper>.empty()` to represent the `notSet` state.